File: code-coaches/easy/land-ho/land_ho.cpp
#include <iostream>
#include <cmath>

int get_waiting_time(int num_people_ahead)
{
    int trip_duration = 10;
    int boat_capacity = 20;

    int num_boats_needed = ceil(
        (double) (num_people_ahead + 1) / boat_capacity
    );  // add 1 to account for yourself

    // trip_duration * 2 because of round trip
    // num_boats_needed - 1 to not include our boat
    int waiting_time = (num_boats_needed - 1) * trip_duration * 2;

    int total_waiting_time = waiting_time + trip_duration;
    return total_waiting_time;
}

int main()
{
    int num_people_ahead;
    std::cin >> num_people_ahead;

    std::cout << get_waiting_time(num_people_ahead);

    return 0;
}

File: code-coaches/easy/land-ho/LandHo.swift
import Foundation

func getWaitingTime(_ numPeopleAhead: Int) -> Int {
    let tripDuration = 10
    let boatCapacity = 20

    // add 1 to account for yourself
    let numBoatsNeeded = Int(ceil(Double(numPeopleAhead + 1) / Double(boatCapacity)))

    // tripDuration * 2 because of round trip
    // numBoatsNeeded - 1 to not include our boat
    let waitingTime = (numBoatsNeeded - 1) * tripDuration * 2

    let totalWaitingTime = waitingTime + tripDuration
    return totalWaitingTime
}

func main() {
    let numPeopleAhead = Int(readLine()!)!
	print(getWaitingTime(numPeopleAhead))
}

main()